v3/src/components/case-table/case-table-tool-shelf-button.tsx

```diff
@@ -1,29 +1,195 @@
+import { createDefaultTileOfType } from "../../models/codap/add-default-content"
+import { kCaseTableTileType } from "./case-table-defs"
+import { getPositionOfNewComponent } from "../../utilities/view-utils"
+import { CodapModal } from "../codap-modal"
+import { uiState } from "../../models/ui-state"
 import TableIcon from "../../assets/icons/icon-table.svg"
 import TrashIcon from "../../assets/icons/icon-trash.svg"
+import AlertIcon from "../../assets/icons/icon-alert.svg"
 
 import "../tool-shelf/tool-shelf.scss"
 
-export const CaseTableToolShelfButton = () => {
-  const data = useDataSetContext()
+const kDefaultTableSize = { width: 186, height: 200 }
+
+export const CaseTableToolShelfMenuList = observer(function CaseTableToolShelfMenuList() {
+  const document = appState.document
+  const content = document.content
+  const manager = getSharedModelManager(document)
+  const datasets = manager?.getSharedModelsByType<typeof SharedDataSet>(kSharedDataSetType)
+  const existingTableTiles = document.content?.getTilesOfType(kCaseTableTileType)
+  const tileIds: string[] = []
+  existingTableTiles?.forEach(tile => tileIds.push(tile.id))
+  const row = content?.getRowByIndex(0)
+  const { isOpen, onOpen, onClose } = useDisclosure()
+  const [modalOpen, setModalOpen] = useState(false)
+  const [dataSetIdToDeleteId, setDataSetIdToDelete] = useState("")
+
+  if (!row) return null
+
+  const handleCreateNewDataSet = () => {
+    const tile = createDefaultTileOfType(kCaseTableTileType)
+    if (!tile) return
+    const newData = [{ AttributeName: "" }]
+    const ds = DataSet.create({ name: "New Dataset" })
+    ds.addAttribute({ name: "AttributeName" })
+    ds.addCases(toCanonical(ds, newData))
+    const { sharedData, caseMetadata } = gDataBroker.addDataSet(ds, tile.id)
+    manager?.addTileSharedModel(tile.content, sharedData, true)
+    manager?.addTileSharedModel(tile.content, caseMetadata, true)
+
+    const width = kDefaultTableSize.width
+    const height = kDefaultTableSize.height
+    const {x, y} = getPositionOfNewComponent({width, height})
+    content?.insertTileInRow(tile, row, {x, y, width, height})
+    uiState.setFocusedTile(tile.id)
+  }
+
+  const openTableForDataset = (model: ISharedDataSet, caseMetadata: ISharedCaseMetadata) => {
+    const tile = createDefaultTileOfType(kCaseTableTileType)
+      if (!tile) return
+      manager?.addTileSharedModel(tile.content, model, true)
+      manager?.addTileSharedModel(tile.content, caseMetadata, true)
+
+      const width = kDefaultTableSize.width
+      const height = kDefaultTableSize.height
+      const {x, y} = getPositionOfNewComponent({width, height})
+      content?.insertTileInRow(tile, row, {x, y, width, height})
+      uiState.setFocusedTile(tile.id)
+  }
 
+  const handleOpenDataSetTable = (dataset: ISharedDataSet) => {
+    const model = manager?.getSharedModelsByType("SharedDataSet").find(m =>  m.id === dataset.id) as ISharedDataSet
```

Comment:
```suggestion
    const model = manager?.getSharedModelsByType("SharedDataSet")
                          .find(m => m.id === dataset.id) as ISharedDataSet | undefined
```
Types should acknowledge the fact that it might not be found.